lnbits/core/views/node_api.py

```diff
@@ -0,0 +1,189 @@
+from typing import List, Optional
+
+import httpx
+from fastapi import APIRouter, Body, Depends, HTTPException
+from pydantic import BaseModel
+from starlette.status import HTTP_503_SERVICE_UNAVAILABLE
+
+from lnbits.decorators import check_admin, parse_filters
+from lnbits.nodes import get_node_class
+from lnbits.settings import settings
+
+from ...db import Filters, Page
+from ...nodes.base import (
+    ChannelPoint,
+    Node,
+    NodeChannel,
+    NodeInfoResponse,
+    NodeInvoice,
+    NodeInvoiceFilters,
+    NodePayment,
+    NodePaymentsFilters,
+    NodePeerInfo,
+    PublicNodeInfo,
+)
+from .. import core_app
+
+
+def require_node():
+    NODE = get_node_class()
+    if not NODE:
+        raise HTTPException(
+            status_code=400,
+            detail="Active backend does not implement Node API",
+        )
+    if not settings.lnbits_node_ui:
+        raise HTTPException(
+            status_code=404,
+            detail="Not enabled",
+        )
+    return NODE
+
+
+def check_public():
+    if not (settings.lnbits_node_ui and settings.lnbits_public_node_ui):
+        raise HTTPException(
+            status_code=404,
+            detail="Not enabled",
+        )
+
+
+node_api = APIRouter(prefix="/node/api/v1", dependencies=[Depends(check_admin)])
+public_node_api = APIRouter(
+    prefix="/node/public/api/v1", dependencies=[Depends(check_public)]
+)
+
+
+@node_api.get(
+    "/ok",
+    description="Check if node api can be enabled",
+    status_code=200,
+    dependencies=[Depends(require_node)],
+)
+async def api_get_ok():
+    pass
+
+
+@public_node_api.get("/info", response_model=PublicNodeInfo)
+async def api_get_public_info(node: Node = Depends(require_node)) -> PublicNodeInfo:
+    return await node.get_public_info()
+
+
+@node_api.get("/info")
+async def api_get_info(
+    node: Node = Depends(require_node),
+) -> Optional[NodeInfoResponse]:
+    return await node.get_info()
+
+
+@node_api.get("/channels")
+async def api_get_channels(
+    node: Node = Depends(require_node),
+) -> Optional[List[NodeChannel]]:
+    return await node.get_channels()
+
+
+@node_api.post("/channels", response_model=ChannelPoint)
+async def api_create_channel(
+    node: Node = Depends(require_node),
+    peer_id: str = Body(),
+    funding_amount: int = Body(),
+    push_amount: Optional[int] = Body(None),
+    fee_rate: Optional[int] = Body(None),
+):
+    return await node.open_channel(peer_id, funding_amount, push_amount, fee_rate)
+
+
+@node_api.delete("/channels")
+async def api_delete_channel(
+    short_id: Optional[str],
+    funding_txid: Optional[str],
+    output_index: Optional[int],
+    force: bool = False,
+    node: Node = Depends(require_node),
```

Comment:
- this is a `DELETE` with no user permission required (anyone can call it if a certain flag is true)?